test/e2e/app/app.go

```diff
@@ -291,7 +331,7 @@ func (app *Application) GetBlob(height int64) ([]byte, bool, error) {
 	}
 	// If not found, reach out to other peers and retrieve it through them.
 	time.Sleep(100 * time.Millisecond) // Add simulated network delay
-	blobFromPeer, exist := blobOracle(height)
+	blobFromPeer, exist := blobOracle(height, app.cfg.BlobMaxBytesUpdateHeight)
```

Comment:
If I understood the logic properly, you shouldn't use `app.cfg.BlobMaxBytesUpdateHeight` here, because that's confusing. For instance, what if the value of `app.cfg.BlobMaxBytes` is 0? Is `app.cfg.BlobMaxBytesUpdateHeight` able to tell you whether there should be a blob at the current height?

IMO, `app.cfg.BlobMaxBytesUpdateHeight` should only be used in `updateBlobMaxBytes`. Everywhere else (e.g. here), you should rely on the value set in the app's DB (either at init time, or by `updateBlobMaxBytes`).

Reply:
It works because right now the only time we return a blob is if `checkBlobHeight` returns true. And this function returns false if :
- We are below `app.cfg.BlobMaxBytesUpdateHeight` 
- The `blobMaxBytes == 0` 

But in this particular case it is not supposed to be passed at all, I did not remove the variable.

Reply:
This can be done, by reading the app state then and checking `MaxBytes`. Because `app.cfg.BlobMaxBytes` is set to the max bytes we want to use *after* the `BlobMaxBytesUpdateHeight`. We cannot thus use it as the source of truth whether blobs should be used and I wanted to avoid parsing the state at every height.